### PR TITLE
Use List instead of ArrayList as input parameter or return type

### PR DESCRIPTION
**Pull request by Maragues**
Solves issue https://github.com/PhilJay/MPAndroidChart/issues/385

I've gone through all the examples and they work ok, except for some that are also failing on your repo, which are

- CombinedChart - throws NPE, mRenderer is not initialized
- PieChart  - hole is draw black instead of transparent
- Realtime chart - throws NPE, mRenderer is not initialized
- Dynamic Data Adding - causes ANR

As I said, I've checked that these errors are also present in your master branch.

### CODE REVIEW

**Comment by Maragues** [comment]
The NPEs in CombinedChart and RealTimeChart may be fixed by PR https://github.com/PhilJay/MPAndroidChart/pull/422

File: MPChartLib/src/com/github/mikephil/charting/data/ChartData.java [hunk 1]
```diff
@@ -95,41 +97,41 @@ public ChartData(ArrayList<String> xVals, ArrayList<T> sets) {
...
 
     /**
-     * Turns an array of strings into an arraylist of strings.
+     * Turns an array of strings into an List of strings.
      * 
      * @param array
      * @return
      */
-    private ArrayList<String> arrayToArrayList(String[] array) {
+    private List<String> arrayToList(String[] array) {
```
Comment (MGaetan89) [inline MPChartLib/src/com/github/mikephil/charting/data/ChartData.java]: `return Arrays.asList(arrays);`?

Comment (Maragues) [inline MPChartLib/src/com/github/mikephil/charting/data/ChartData.java]: Yeah, that'd make much more sense. I didn't go into analyzing improvements or if the code makes sense. Thanks!

**Comment by PhilJay** [comment]
Thanks for this PR, unfortunately I cannot merge it because it interfears with a lot of changes I recently made.

Best regards,
Phil

**Comment by friederbluemle** [comment]
This would still be nice to have, as it's generally better to use the interface of a collection type rather than a specific implementation. Maybe the PR can be rebased and reopened?

**Comment by Maragues** [comment]
Thanks @friederbluemle , I was feeling quite alone :-) 

I can merge from upstream next monday and re-do the changes. Are you ok with that @PhilJay ?

**Comment by Maragues** [comment]
@PhilJay I merged your master branch into mine, every example is working fine. Would you like to reopen this PR or should I create a new one? If you reopen it, I don't know if it'll include my latest commit.

**Comment by PhilJay** [comment]
Thanks, just create a new one!
So now the lib takes list instead of arraylist as a parameter?

**Comment by Maragues** [comment]
Yes, it'll accept any Object that implements List. In principle that doesn't change anything, since you store the index in every Entry instance, but the API is now more flexible and doesn't impose the ArrayList constraint.

**Pull request closed without merging**